src/components/smart_objects/src/smart_object.cc

```diff
@@ -723,16 +723,28 @@ void SmartObject::duplicate(const SmartObject& OtherObject) {
 void SmartObject::cleanup_data() {
   switch (m_type) {
     case SmartType_String:
-      delete m_data.str_value;
+      if (m_data.str_value) {
```

Comment:
@AGaliuzov You know, in C plus, if value in ```if``` condition is ```NULL```, operator ```delete``` should do nothing, check the value that to delete is a good style in C plus. And Also, there is a very small possibility that delete fails, that may cause memory leak. Third, this is to avoid of dangling pointer.